ext/standard/head.c

```diff
@@ -53,13 +55,15 @@ PHP_FUNCTION(header)
 PHP_FUNCTION(header_remove)
 {
 	sapi_header_line ctr = {0};
+	char *line = NULL;
 	size_t len = 0;
 
 	ZEND_PARSE_PARAMETERS_START(0, 1)
 		Z_PARAM_OPTIONAL
-		Z_PARAM_STRING(ctr.line, len)
+		Z_PARAM_STRING(line, len)
```

Comment:
This makes me wonder why the Z_PARAM_STRING target is `char **` and not `const char **`. Maybe we should change that as well (but separately -- it may have larger fallout).

Reply:
I couldn't agree more, that's what I thought a long time ago
but if we change it, it may break many PECL extensions. and we can't reflect it is read-only when we use `Z_PARAM_STR`

Reply:
Yeah, I'm not sure what to do about this one. With the old zend_parse_parameters API, it was possible to use both const char and char, because it simply wasn't validated. With Z_PARAM_STRING it is, and the wrong one has been originally used :(

For the zend_string APIs, I don't think "const" concept makes a lot of sense, because semantics are determined by copy-on-write. Effectively zend_string is "always" const (apart from refcount changes).

Reply:
I think it may be appropriate to fix it in PHP8 because we already have a lot of BC breaks 🤪...